api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -125,20 +125,24 @@ public static Schema select(Schema schema, Set<Integer> fieldIds) {
     return ImmutableSet.copyOf(getIdsInternal(type));
   }
 
+  private static Set<Integer> getIdsInternal(Type type, boolean includeStructIds) {
+    return visit(type, new GetProjectedIds(includeStructIds));
+  }
+
   private static Set<Integer> getIdsInternal(Type type) {
-    return visit(type, new GetProjectedIds());
+    return getIdsInternal(type, true);
```

Comment:
Is it a lot of changes to remove this method? I wouldn't expect it since this is `private`. I'd probably prefer removing it, but up to you if it touches a bunch of unrelated code.

Reply:
We are pretty safe here, removing it. Only two changes in this file